### ISSUE TITLE
Timers not being removed from "sorted_all" attribute on next_timer entity, it keeps growing

### ISSUE THREAD

**Issue by dpgh947** [custom-components/alexa_media_player#1936]
**IMPORTANT: Please search the issues, including closed issues, and the [FAQ](https://github.com/custom-components/alexa_media_player/wiki/FAQ) before opening a new issue. The template is mandatory; failure to use it will result in issue closure.**

**Describe the bug**

The sorted_all attribute on the ...next_timer entities has started growing because finished or cancelled timers are now staying there. I noticed this first time yesterday (May 7th) when I had an error logged saying the attributes on one of my devices had gone over 16kb.

**To Reproduce**
Display the "...next_timer" entity for a device in developer tools. Set a timer on that device, data for it appears under "sorted_active" and "sorted_all", and "total_all" is incremented. Allow the timer to end, or cancel it, and it remains there under "sorted_all", and total_all remains unchanged. Set another timer, it gets added to the sorted_all list, and so on. sorted_all now just keeps growing with every timer set.

I have had varying results when removing the data under developer tools using "set state" - sometimes the change seems to stick and the big list does not come back when setting another timer, and sometimes it does.

**Expected behavior**

I have not paid much attention to these attributes before and I don't see why there is an "active" and an "all" list, but the "all" list shouldn't just grow forever, which is what happened to me yesterday when I was setting lots of timers while cooking a big lunch. 

**Screenshots**

**System details**

- Home-assistant (version): 2023.4.6
- alexa_media (version from `const.py` or HA startup): 4.6.2 and just now 4.6.4
- alexapy (version from `pip show alexapy` or HA startup):
- Amazon [2FA is enabled](https://github.com/custom-components/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account) (y/n). : y

**Logs**
Please provide [logs](https://github.com/custom-components/alexa_media_player/wiki/FAQ#how-do-i-enable-debug-logging-for-the-component).

**Additional context**

**Comment by dpgh947** [comment]
I wonder if this is an api problem at amazon... If I set a timer, and then from the phone app pause it and then delete it, it disappears from the entity completely as might be expected. If I set it and then cancel it by voice on the device, it remains there under sorted_all. As for getting rid of the attribute when it has grown huge, if I edit it under developer tools and then do nothing else with any alexas for a while, the delete seems to stick. If I create another timer soon after removing it, all the previous data comes back. In other words, it is very hard to recreate consistent results every time I do something, it keeps varying. But I have never seen this "attribute greater than 16kb" before and I have had several yesterday and today, so something has changed/temporarily gone wrong somewhere.
